crypto/evp/evp_lib.c

```diff
@@ -648,18 +648,26 @@ int EVP_CIPHER_get_key_length(const EVP_CIPHER *cipher)
 int EVP_CIPHER_CTX_get_key_length(const EVP_CIPHER_CTX *ctx)
 {
     if (ctx->key_len <= 0 && ctx->cipher->prov != NULL) {
-        /*-
-         * Casting away the const is annoying but required here.  We need to
-         * cache the result for performance reasons.
-         */
-        int *const len = &((EVP_CIPHER_CTX *)ctx)->key_len;
         int ok;
         OSSL_PARAM params[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
+        size_t len;
 
-        params[0] = OSSL_PARAM_construct_int(OSSL_CIPHER_PARAM_KEYLEN, len);
+        params[0] = OSSL_PARAM_construct_size_t(OSSL_CIPHER_PARAM_KEYLEN, &len);
         ok = evp_do_ciph_ctx_getparams(ctx->cipher, ctx->algctx, params);
         if (ok <= 0)
             return EVP_CTRL_RET_UNSUPPORTED;
+
+        /* Should never happen since EVP_MAX_KEY_LENGTH is less than INT_MAX */
+        if (len > INT_MAX) {
+            ERR_raise(ERR_LIB_EVP,
```

Comment:
You would have to use ERR_LIB_CRYPTO here. Or use EVP_R_INVALID_KEY_LENGTH.

Reply:
It's not an invalid key length, the provider thinks it's fine but it's out of range.

I've reworked this to avoid the need to raise an error here.